internal/data/common.go

```diff
@@ -109,30 +117,52 @@ func (a AttributeValue) BoolVal() bool {
 }
 
 func (a AttributeValue) SetString(v string) {
-	a.orig.Type = otlpcommon.AttributeKeyValue_STRING
+	a.setType(otlpcommon.AttributeKeyValue_STRING)
```

Comment:
👍 
Otherwise we would keep all fields around and pollute the wire when marshaling.